lib/useOnyx.ts

```diff
@@ -133,24 +131,29 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(key: TKey
             newFetchStatus = 'loading';
         }
 
-        // If data is not present in cache (if it's `undefined`) and `initialValue` is set during the first connection,
+        // If `options.initWithStoredValues` is explicitly set to `false`, the fetch status should always be `loaded`.
+        if (options?.initWithStoredValues === false) {
```

Comment:
Is this only applicable during the first connection or all the time? If only the first time we should use `isFirstConnectionRef.current`.

```ts
        if (isFirstConnectionRef.current && options?.initWithStoredValues === false) {
```

Reply:
we can actually remove this.